packages/dataviews/src/types.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+type Item = Record< string, any >;
+
+interface Option {
+	value: any;
+	label: string;
+}
+
+interface filterByConfig {
+	operators?: string[];
+	isPrimary?: boolean;
+}
+
+export interface Field {
+	/**
+	 * The unique identifier of the field.
+	 */
+	id: string;
+
+	/**
+	 * The label of the field. Defaults to the id.
+	 */
+	header?: string;
+
+	/**
+	 * Callback used to retrieve the value of the field from the item.
+	 * Defaults to `item[ field.id ]`.
+	 */
+	getValue?: ( { item }: { item: Item } ) => any;
+
+	/**
+	 * Callback used to render the field. Defaults to `field.getValue`.
+	 */
+	render?: ( { item }: { item: Item } ) => ReactNode;
+
+	/**
+	 * The width of the field column.
+	 */
+	width?: string | number;
+
+	/**
+	 * The minimum width of the field column.
+	 */
+	maxWidth?: string | number;
+
+	/**
+	 * The maximum width of the field column.
+	 */
+	minWidth?: string | number;
+
+	/**
+	 * Whether the field is sortable.
+	 */
+	enableSorting?: boolean;
+
+	/**
+	 * Whether the field is searchable.
+	 */
+	enableGlobalSearch?: boolean;
+
+	/**
+	 * Whether the field is filterable.
+	 */
+	enableHiding?: boolean;
+
+	/**
+	 * The type of the field.
+	 */
+	type?: string;
+
+	/**
+	 * The list of options to pick from when using the field as a filter.
+	 */
+	elements?: Option[];
+
+	/**
+	 * Filter config for the field.
+	 */
+	filterBy?: filterByConfig;
```

Comment:
Should elements be part of "filterBy". Also the "filterBy" property is a bit unclear to be honest. Anyway, something to improve separately.

Reply:
The idea for `elements` is that it declares what's the accepted values for the fields, so it can be shared by the editing and filtering functions. We don't have any edit capabilities for testing this how well this fare, though.